test/ae_database/databasesetup.php

```diff
@@ -0,0 +1,254 @@
+<?php
+include "AEData.inc";
+
+sqlsrv_configure( 'WarningsReturnAsErrors', 1 );
+sqlsrv_configure( 'LogSeverity', SQLSRV_LOG_SEVERITY_ALL );  
+
+$databaseName = 'AEDemo';
+$uid = 'yourUsername';
+$pwd = 'yourPassword';
+$server = 'yourServer';
+
+$connectionInfo = array( "Database"=>$databaseName, "UID"=>$uid, "PWD"=>$pwd);
+
+$conn = sqlsrv_connect( $server, $connectionInfo );
+if( $conn === false )
+{
+    echo "Failed to connect.\n";
+    die( print_r(sqlsrv_errors(), true ));
+}
+
+// create table for exact numerics
+$stmt = sqlsrv_query($conn, "IF OBJECT_ID('dbo.test_AE_exnum', 'U') IS NOT NULL DROP TABLE [dbo].[test_AE_exnum]");
+if ($stmt === false) {
+    echo "Failed to drop table.\n";
+    die(print_r(sqlsrv_errors(), true));
+}
+
+$sql_create = "CREATE TABLE dbo.test_AE_exnum([normBigint] [bigint], [encDetBigint] [bigint], [encRandBigint] [bigint],
+                                                       [normInt] [int], [encDetInt] [int], [encRandInt] [int],
+                                                       [normSmallint] [smallint], [encDetSmallint] [smallint], [encRandSmallint] [smallint],
+                                                       [normTinyint] [tinyint], [encDetTinyint] [tinyint], [encRandTinyint] [tinyint],
+                                                       [normDecimal] [decimal](18,5), [encDetDecimal] [decimal](18,5), [encRandDecimal] [decimal](18,5),
+                                                       [normNumeric] [numeric](18,5), [encDetNumeric] [numeric](18,5), [encRandNumeric] [numeric](18,5),
+                                                       [normMoney] [money], [encDetMoney] [money], [encRandMoney] [money],
+                                                       [normSmallmoney] [smallmoney], [encDetSmallmoney] [smallmoney], [encRandSmallmoney] [smallmoney],
+                                                       [normBit] [bit], [encDetBit] [bit], [encRandBit] [bit])";
+
+$stmt = sqlsrv_query($conn, $sql_create);
+if ($stmt === false) {
+    echo "Failed to create table.\n";
+    die(print_r(sqlsrv_errors(), true));
```

Comment:
Since this setup script contains more than one table, consider making all the error messages more descriptive. For example, here it should be 'failed to create table test_AE_exnum' or something like that. When errors occurred we will know what exactly went wrong.